client.py

```diff
@@ -247,42 +258,51 @@ def _close(self):
                 serv = socket.getaddrinfo(self._server, self._port)[0][-1]  # server read
                 # If server is down OSError e.args[0] = 111 ECONNREFUSED
                 self._sock.connect(serv)
+            except OSError as e:
```

Comment:
This catches connection errors to determine if the server is available or got rebooted.
When sending the id later on, it is obvious that the server is available but there could be a different program running on that port. This might seem to detailed and unimportant to you an can certainly be removed again.